server/cli/kv.go

```diff
@@ -0,0 +1,220 @@
+	Short:     "sets the value for a key",
+	Long: `
+Sets the value for one or more keys. Keys and values must be provided
+in pairs on the command line. All of the key/value pairs are set within
+a transaction.
+`,
+	Run:  runPut,
+	Flag: *flag.CommandLine,
+}
+
+func runPut(cmd *commander.Command, args []string) {
+	if len(args) == 0 || len(args)%2 == 1 {
+		cmd.Usage()
+		return
+	}
+
+	// Do not allow system keys to be put.
+	for i := 0; i < len(args); i += 2 {
+		if strings.HasPrefix(args[i], "\x00") {
+			fmt.Fprintf(os.Stderr, "unable to put system key: %s\n", proto.Key(args[i]))
+			os.Exit(1)
+		}
+	}
+
+	kv := makeKVClient()
+	defer kv.Close()
+
+	opts := &client.TransactionOptions{Name: "test", Isolation: proto.SERIALIZABLE}
+	err := kv.RunTransaction(opts, func(txn *client.KV) error {
+		for i := 0; i < len(args); i += 2 {
+			key := proto.Key(args[i])
+			value := []byte(args[i+1])
+			txn.Prepare(proto.Put, proto.PutArgs(key, value), &proto.PutResponse{})
+		}
+		return nil
+	})
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "put failed: %s\n", err)
+		os.Exit(1)
+	}
+}
+
+// A CmdPut command sets the value for one or more keys.
+var CmdDel = &commander.Command{
+	UsageLine: "del [options] <key> [<key2>...]",
+	Short:     "deletes the value for a key",
+	Long: `
+Deletes the value for one or more keys.
+`,
+	Run:  runDel,
+	Flag: *flag.CommandLine,
+}
+
+func runDel(cmd *commander.Command, args []string) {
+	if len(args) == 0 {
+		cmd.Usage()
+		return
+	}
+
+	// Do not allow system keys to be deleted.
+	for i := 0; i < len(args); i++ {
+		if strings.HasPrefix(args[i], "\x00") {
+			fmt.Fprintf(os.Stderr, "unable to delete system key: %s\n", proto.Key(args[i]))
+			os.Exit(1)
+		}
+	}
+
+	kv := makeKVClient()
+	defer kv.Close()
+
+	opts := &client.TransactionOptions{Name: "test", Isolation: proto.SERIALIZABLE}
+	err := kv.RunTransaction(opts, func(txn *client.KV) error {
+		for i := 0; i < len(args); i++ {
+			key := proto.Key(args[i])
+			txn.Prepare(proto.Delete, proto.DeleteArgs(key), &proto.DeleteResponse{})
+		}
+		return nil
+	})
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "delete failed: %s\n", err)
+		os.Exit(1)
+	}
+}
+
+// A CmdScan command fetches the key/value pairs for a specified
+// range.
+var CmdScan = &commander.Command{
+	UsageLine: "scan [options] [<start-key> [<end-key>]]",
+	Short:     "scans a range of keys",
+	Long: `
+Fetches and display the key/value pairs for a range. If no <start-key>
+is specified then all (non-system) key/value pairs are retrieved. If no
+<end-key> is specified then all keys greater than or equal to <start-key>
+are retrieved.
+
+Caveat: Currently only retrieves 1000 keys.
+`,
+	Run:  runScan,
+	Flag: *flag.CommandLine,
+}
+
+func runScan(cmd *commander.Command, args []string) {
+	if len(args) > 2 {
+		cmd.Usage()
+		return
+	}
+	var (
+		startKey proto.Key
+		endKey   proto.Key
+	)
+	if len(args) >= 1 {
+		startKey = proto.Key(args[0])
```

Comment:
Should we disallow start keys that are empty or start with `\x00` to prevent access to system keys or allow it on the assumption that the user knows what they're doing if they ask for that?